_dasldemos/computational_rank.md

```diff
@@ -0,0 +1,200 @@
+4. [**Do I build my package from source?**](#4-best-practices-when-compiling-from-source) This should be your last resort, because it often requires knowledge of compilers to get it up and running. You will probably be using some variant of `make`.
+
+The next section talks about basic best practices when running software using each of these methods.
+
+## General Advice
+
+- Before reinventing the wheel, check out the [WILDS Docker Repository](https://getwilds.org/packages.html#docker) to see if there’s a similar workflow to what you need. You may just need to [fork an existing workflow](https://getwilds.org/packages.html#wdls) (make your own version) to get started.
+- Avoid mixing and matching your software environments and loading them all at once to avoid dependency conflicts. In practice, this means sticking to either easymodules or Docker containers if possible.
+- Pick a version number for your software. This practice will keep your workflow reproducible if the latest version of the software changes. For easymodules, you can see the versions using `module avail <module-name>`, such as `module avail samtools`.
+- For Docker containers, pick a tagged container from the container image library. For example:  [`biocontainers/samtools:v1.9-4-deb_cv1`](https://hub.docker.com/layers/biocontainers/samtools/v1.9-4-deb_cv1/images/sha256-da61624fda230e94867c9429ca1112e1e77c24e500b52dfc84eaf2f5820b4a2a?context=explore)
+
+## 1. Best Practices when using easymodules
+
+### Finding modules
+
+1. Documentation on `lmod` (what you use when you do `module load` is [here](https://lmod.readthedocs.io/en/latest/010_user.html))
+2. Find available modules using either [the webpages](https://sciwiki.fredhutch.org/scicomputing/compute_scientificSoftware/) or `module avail`. Find a specific version number (if you don’t know, pick the latest version). For example: `module avail samtools` 
+
+### In your script
+
+1. [The first line should be to load the module script](https://sciwiki.fredhutch.org/scicomputing/compute_environments/#best-practice-1): `source /app/lmod/lmod/init/profile`
+2. `module load` when about to use software in your script (use specific version numbers). For example: `module load SAMtools/0.1.20-foss-2018b` 
+3. As a dry run on the command line, `module load` and check to see that you can see the executable using `which` . For example: `which samtools` should return the path where this binary is installed (`/app/software/SAMtools/0.1.20-foss-2018b/bin/samtools`).
+4. Do you work in the script. For example: `samtools view -c myfile_bam > counts.txt`
+5. `module unload` your module or `module purge` all modules when done using them and starting a new step. Try not to load all modules all at once at beginning of script. For example: `module unload SAMtools/0.1.20-foss-2018b` 
+
+Here’s an example illustrating this:
```

Comment:
I think an example is missing here... need one?

Reply:
Yes, I'll make sure this is added.